azure-pipelines.docker.yml

```diff
@@ -0,0 +1,140 @@
+
+resources:
+  containers:
+  - container: devito-pip-gcc49
+    image: 'tmbgreaves/devito-pip-gcc4.9'
+  - container: devito-conda-gcc49
+    image: 'tmbgreaves/devito-conda-gcc4.9'
+  - container: devito-conda-gcc5
+    image: 'tmbgreaves/devito-conda-gcc5'
+  - container: devito-conda-gcc7
+    image: 'tmbgreaves/devito-conda-gcc7'
+  - container: devito-yask-conda-gcc7
+    image: 'tmbgreaves/devito-yask-conda-gcc7'
+  - container: devito-conda-gcc8
+    image: 'tmbgreaves/devito-conda-gcc8'
+
+jobs:
+- job: Container_Build
+  displayName: 'Build using'
+  pool:
+    vmImage: 'Ubuntu 16.04'
+  strategy:
+    matrix:
+      pip-gcc4:
+        gccVersion: '4.9'
+        installMethod: 'pip'
+        installYask: 'false'
+        imageBaseName: 'devito'
+      conda-gcc4:
+        gccVersion: '4.9'
+        installMethod: 'conda'
+        installYask: 'false'
+        imageBaseName: 'devito'
+      conda-gcc5:
+        gccVersion: '5'
+        installMethod: 'conda'
+        installYask: 'false'
+        imageBaseName: 'devito'
+      conda-gcc7:
+        gccVersion: '7'
+        installMethod: 'conda'
+        installYask: 'false'
+        imageBaseName: 'devito'
+      yask-conda-gcc7:
+        gccVersion: '7'
+        installMethod: 'conda'
+        installYask: 'true'
+        imageBaseName: 'devito-yask'
+      conda-gcc8:
+        gccVersion: '8'
+        installMethod: 'conda'
+        installYask: 'false'
+        imageBaseName: 'devito'
+
+  variables:
+     imageName: $(imageBaseName)-$(installMethod)-gcc$(gccVersion)
+     dockerFile: Dockerfile.pipelines
+     
+  steps:
+  - script: |
+      docker build -f $(dockerFile) -t $(dockerId)/$(imageName):$(Build.BuildId) --build-arg gccvers=$(gccVersion) --build-arg installMethod=$(installMethod) --build-arg yask=$(installYask) .
+      docker tag $(dockerId)/$(imageName):$(Build.BuildId) $(dockerId)/$(imageName):latest
+      docker login -u $(dockerId) -p $(dockerPassword)
+      docker push $(dockerId)/$(imageName):$(Build.BuildId)
+      docker push $(dockerId)/$(imageName):latest
+
+
+- job: Run_Tests
+  displayName: 'Test using'
+  dependsOn: Container_Build
+  condition: succeeded('Container_Build')
+  pool:
+    vmImage: 'Ubuntu 16.04'
+  strategy:
+    matrix:
+      pip-gcc49:
+        containerName: devito-pip-gcc49
+        testMethod: 'pip'
+      openmp-conda-gcc49:
+        containerName: devito-pip-gcc49
+        testMethod: 'conda'
+        useOpenMP: 'true'
+      conda-gcc5: 
+        containerName: devito-conda-gcc5
+        testMethod: 'conda'
+      yask-conda-gcc7:
+        containerName: devito-yask-conda-gcc7
+        testMethod: 'conda'
+        devitoBackend: 'yask'
+      ops-conda-gcc7:
+        containerName: devito-conda-gcc7
+        testMethod: 'conda'
+        devitoBackend: 'OPS'
+      conda-gcc8: 
+        containerName: devito-conda-gcc8
+        testMethod: 'conda'
+
+  container: $[ variables['containerName'] ]
+  timeoutInMinutes: 180
```

Comment:
I suppose we can lower it to 120 ?

Reply:
Yes, I think the default 60 is in the right ballpark for the expected runtime on the free Pipelines nodes, so 120 should be very safe.